www/js/lib/zstdec_wrapper.js

```diff
@@ -0,0 +1,260 @@
+﻿/**
+ * zstdec_wrapper.js: Javascript wrapper around compiled zstd decompressor.
+ *
+ * Copyright 2020 Jaifroid, Mossroy and contributors
+ * License GPL v3:
+ *
+ * This file is part of Kiwix.
+ *
+ * Kiwix is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Kiwix is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Kiwix (file LICENSE-GPLv3.txt).  If not, see <http://www.gnu.org/licenses/>
+ */
+'use strict';
+define(['q', 'zstdec'], function(Q) {
+    // DEV: zstdec.js has been compiled with `-s EXPORT_NAME="ZD" -s MODULARIZE=1` to avoid a clash with xzdec which uses "Module" as its exported object
+    // Note that we include zstdec above in requireJS definition, but we cannot change the name in the function list
+    // There is no longer any need to load it in index.html
+    // For explanation of loading method below to avoid conflicts, see https://github.com/emscripten-core/emscripten/blob/master/src/settings.js
+    
+    /**
+     * The ZSTD Decoder instance
+     * @constructor Constructs the zd object representing a ZSTD decoder Emscripten instance
```

Comment:
I don't see which constructor this corresponds to

Reply:
It was intended to be the then function below, which instantiates the `zd` instance (object) and adds a whole load of needed properties (deocumented here) to the the object.  But I don't know technical definitions, so maybe this doesn't qualify as an object constructor?

Reply:
I suppose the JSDoc tags should document the code of the current file.
Here you're documenting the `instance` parameter and/or the zd variable.
Maybe you could define a zd type with `@typedef`, even if this type is not used as a parameter (and thus I suppose the IDE won't use it for autocompletion), and add a comment line to explain this describes the structure of zd variable